crates/ruff_linter/src/rules/ruff/rules/class_with_mixed_type_vars.rs

```diff
@@ -0,0 +1,243 @@
+/// - [Python documentation: type parameter lists](https://docs.python.org/3/reference/compound_stmts.html#type-params)
+/// - [PEP 695 - Type Parameter Syntax](https://peps.python.org/pep-0695/)
+///
+/// [PEP 695]: https://peps.python.org/pep-0695/
+/// [type parameter lists]: https://docs.python.org/3/reference/compound_stmts.html#type-params
+#[derive(ViolationMetadata)]
+pub(crate) struct ClassWithMixedTypeVars;
+
+impl Violation for ClassWithMixedTypeVars {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "Class with type parameter list inherits from `Generic`".to_string()
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some("Remove `Generic` base class".to_string())
+    }
+}
+
+/// RUF060
+pub(crate) fn class_with_mixed_type_vars(checker: &mut Checker, class_def: &StmtClassDef) {
+    if checker.settings.target_version < PythonVersion::Py312 {
+        return;
+    }
+
+    let StmtClassDef {
+        type_params,
+        arguments,
+        ..
+    } = class_def;
+
+    let Some(type_params) = type_params else {
+        return;
+    };
+
+    let Some(arguments) = arguments else {
+        return;
+    };
+
+    let Some((generic_base, old_style_type_vars)) =
+        typing_generic_base_and_arguments(arguments, checker.semantic())
+    else {
+        return;
+    };
+
+    let mut diagnostic = Diagnostic::new(ClassWithMixedTypeVars, generic_base.range);
+
+    if let Some(fix) = convert_type_vars(
+        generic_base,
+        old_style_type_vars,
+        type_params,
+        arguments,
+        checker,
+    ) {
+        diagnostic.set_fix(fix);
+    }
+
+    checker.diagnostics.push(diagnostic);
+}
+
+fn typing_generic_base_and_arguments<'a>(
+    class_arguments: &'a Arguments,
+    semantic: &SemanticModel,
+) -> Option<(&'a ExprSubscript, &'a Expr)> {
+    let (_, base @ ExprSubscript { slice, .. }) = find_generic(class_arguments, semantic)?;
+
+    Some((base, slice.as_ref()))
+}
+
+fn convert_type_vars(
+    generic_base: &ExprSubscript,
+    old_style_type_vars: &Expr,
+    type_params: &TypeParams,
+    class_arguments: &Arguments,
+    checker: &Checker,
+) -> Option<Fix> {
+    let mut type_vars = type_params
+        .type_params
+        .iter()
+        .map(TypeVar::from)
+        .collect::<Vec<_>>();
+
+    let semantic = checker.semantic();
+    let mut converted_type_vars = match old_style_type_vars {
+        Expr::Tuple(ExprTuple { elts, .. }) => {
+            generic_arguments_to_type_vars(elts.iter(), type_params, semantic)?
+        }
+        expr @ (Expr::Subscript(_) | Expr::Name(_)) => {
+            generic_arguments_to_type_vars(iter::once(expr), type_params, semantic)?
+        }
+        _ => return None,
+    };
+
+    type_vars.append(&mut converted_type_vars);
```

Comment:
There's no need to declare the variable as `mut` if you use `extend()` rather than `append`:

```suggestion
    let converted_type_vars = match old_style_type_vars {
        Expr::Tuple(ExprTuple { elts, .. }) => {
            generic_arguments_to_type_vars(elts.iter(), type_params, semantic)?
        }
        expr @ (Expr::Subscript(_) | Expr::Name(_)) => {
            generic_arguments_to_type_vars(iter::once(expr), type_params, semantic)?
        }
        _ => return None,
    };

    type_vars.extend(converted_type_vars);
```